jooby/src/main/java/io/jooby/spi/ValueConverter.java

```diff
@@ -0,0 +1,35 @@
+package io.jooby.spi;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import io.jooby.TypeMismatchException;
+import io.jooby.Value;
+
+/**
+ * An SPI for value conversion.
+ * @author agentgt
+ */
+public interface ValueConverter {
+  /**
+   * A short circuit to see if the converter supports the given type.
+   *
+   * This defaults to true to allow a functional interface since convert can return null
+   * to indicate it does not support the type.
+   *
+   * @param type class or interface
+   * @return true if the converter can convert for the type
+   */
+  default boolean supportsType(@Nonnull Class<?> type) {
+    return true;
```

Comment:
feel it shouldn't be a default method just to get a functional interface. Also, convert should not returns null as on unsupported type. Null should be a valid response type think

Reply:
Damn I thought you would like that. I originally wasn’t going to bring it seems all the cool kids want lambda for everything.

I also was going to use Optional instead of null.

I don’t really have strong opinions either way on it but list like strategies like this often need a way to signal they don’t want to do the work even if they said they support the type.

Reply:
Also it appeared where resolve happens null is not valid but I need to recheck.

Reply:
I'm basically replacing the logic before `newInstance` which cannot return null and thus is not a valid response type. If you allow null you would be changing existing behavior. I'm resolving this for now.